src/components/liveSession/components/liveSessionSubjectModal/liveSessionSubjectModal.component.js

```diff
@@ -9,17 +9,23 @@
             },
             templateUrl: 'components/liveSession/components/liveSessionSubjectModal/liveSessionSubjectModal.template.html',
             controllerAs: 'vm',
-            controller: function ($mdDialog, LiveSessionSubjectSrv, LiveSessionSrv) {
+            controller: function (ENV, $mdDialog, LiveSessionSubjectSrv, LiveSessionSrv, ZnkLessonNotesSrv) {
                 'ngInject';
 
+                 let liveSessionSettingsProm = ZnkLessonNotesSrv.getLiveSessionSettings();
+
                 this.$onInit = () =>  {
                     this.sessionSubjects = LiveSessionSubjectSrv.getLiveSessionTopics();
                     this.closeModal = $mdDialog.cancel;
                 };
 
                 this.startSession = (sessionSubject) => {
-                    let lessonData = { sessionSubject: sessionSubject };
-                    LiveSessionSrv.startLiveSession(this.student, lessonData);
+                    liveSessionSettingsProm.then(liveSessionSettings => {
+                        let expectedSessionEndTime = liveSessionSettings.length || ENV.liveSession.sessionLength;
```

Comment:
Should be "now + liveSessionSettings.length || now + ENV.liveSession.sessionLength